Drop libva-amdgpu for RHEL 9+ and SLE

Mystro256 opened the pull request:
Due to limitations in RPM packaging, having libva-amdgpu packages just causes bugs due to edge cases. It's easier to just use the inbox packages for RHEL9+ and SLE.

I also noticed that SLE should be pulling in libva-drm2 as a runtime requirement.

SWDEV-555510

Note: I have not fully tested this, I will need some help to make sure the python script is good.

Mystro256 committed:
Drop libva-amdgpu for newer distros

Due to bugs in mixing old and new libva installed, having both
libva-amdgpu packages just causes bugs due to edge cases. It's easier
to just use the inbox packages for RHEL9+, SLE, and UB24.04.

RHEL 8 and UB22.04 still needs libva-amdgpu since libva is too old.

I also noticed that SLE should be pulling in libva-drm2 as a runtime
requirement, since rocdecode only uses the DRM interface.

SWDEV-555510
SWDEV-557842

Signed-off-by: Jeremy Newton <Jeremy.Newton@amd.com>

rrawther reviewed (approved):
@AryanSalmanpour : please test and verify these changes

kiritigowda commented:
@Mystro256 need your help resolving a similar issue on Debian 13 

```
vainfo --display drm
Trying display: drm
libva info: VA-API version 1.22.0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/radeonsi_drv_video.so
libva info: Found init function __vaDriverInit_1_22
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.22 (libva 2.22.0)
vainfo: Driver version: Mesa Gallium driver 25.0.7-2 for AMD Radeon RX 7900 XT (radeonsi, navi31, LLVM 19.1.7, DRM 3.64, 6.8.0-83-generic)
vainfo: Supported profile and entrypoints
      VAProfileH264ConstrainedBaseline:	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointEncSlice
      VAProfileH264Main               :	VAEntrypointVLD
      VAProfileH264Main               :	VAEntrypointEncSlice
      VAProfileH264High               :	VAEntrypointVLD
      VAProfileH264High               :	VAEntrypointEncSlice
      VAProfileHEVCMain               :	VAEntrypointVLD
      VAProfileHEVCMain               :	VAEntrypointEncSlice
      VAProfileHEVCMain10             :	VAEntrypointVLD
      VAProfileHEVCMain10             :	VAEntrypointEncSlice
      VAProfileJPEGBaseline           :	VAEntrypointVLD
      VAProfileVP9Profile0            :	VAEntrypointVLD
      VAProfileVP9Profile2            :	VAEntrypointVLD
      VAProfileAV1Profile0            :	VAEntrypointVLD
      VAProfileAV1Profile0            :	VAEntrypointEncSlice
      VAProfileNone                   :	VAEntrypointVideoProc
root@santiago:~# cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 13 (trixie)"
NAME="Debian GNU/Linux"
VERSION_ID="13"
VERSION="13 (trixie)"
VERSION_CODENAME=trixie
DEBIAN_VERSION_FULL=13.1
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

```

kiritigowda commented:
Similar issue on Ubuntu 24, here we ship VA 1.16

```
vainfo --display drm
libva info: VA-API version 1.20.0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/radeonsi_drv_video.so
libva info: Found init function __vaDriverInit_1_20
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.20 (libva 2.12.0)
vainfo: Driver version: Mesa Gallium driver 25.0.7-0ubuntu0.24.04.2 for AMD Radeon RX 7900 XT (radeonsi, navi31, LLVM 20.1.2, DRM 3.64, 6.8.0-83-generic)
vainfo: Supported profile and entrypoints
      VAProfileH264ConstrainedBaseline:	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointEncSlice
      VAProfileH264Main               :	VAEntrypointVLD
      VAProfileH264Main               :	VAEntrypointEncSlice
      VAProfileH264High               :	VAEntrypointVLD
      VAProfileH264High               :	VAEntrypointEncSlice
      VAProfileHEVCMain               :	VAEntrypointVLD
      VAProfileHEVCMain               :	VAEntrypointEncSlice
      VAProfileHEVCMain10             :	VAEntrypointVLD
      VAProfileHEVCMain10             :	VAEntrypointEncSlice
      VAProfileJPEGBaseline           :	VAEntrypointVLD
      VAProfileVP9Profile0            :	VAEntrypointVLD
      VAProfileVP9Profile2            :	VAEntrypointVLD
      VAProfileAV1Profile0            :	VAEntrypointVLD
      VAProfileAV1Profile0            :	VAEntrypointEncSlice
      VAProfileNone                   :	VAEntrypointVideoProc
root@santiago:~# cat /etc/os-release 
PRETTY_NAME="Ubuntu 24.04.1 LTS"
NAME="Ubuntu"
VERSION_ID="24.04"
VERSION="24.04.1 LTS (Noble Numbat)"
VERSION_CODENAME=noble
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=noble
LOGO=ubuntu-logo

```

Mystro256 commented:
Ok I'll change it to include 24.04 as well.

Mystro256 commented:
Updated, I haven't tested too well, so I'll need some help verifying I did it correctly.

kiritigowda committed:
rocDecode-setup.py - Fix

```
rocDecode-setup.py", line 246
    else
        ^
SyntaxError: expected ':'

```

kiritigowda committed:
rocDecode-setup.py - Fix

```
rocDecode-setup.py", line 246
    else
        ^
SyntaxError: expected ':'

```

kiritigowda committed:
rocDecode-setup.py - fix

```
E: Unable to locate package rocm-hip-runtime-devlibstdc++-12-dev
E: Couldn't find any package by regex 'rocm-hip-runtime-devlibstdc++-12-dev'

```

kiritigowda commented:
* rocdecode
```
dpkg -I rocdecode_1.4.0.70002-local_amd64.deb 
 new debian package, version 2.0.
 size 305264 bytes: control archive=549 bytes.
     429 bytes,    11 lines      control              
     232 bytes,     3 lines      md5sums              
 Architecture: amd64
 Depends:  rocm-hip-runtime, libva-drm2 (>= 2.16.0) | libva-amdgpu-drm2, mesa-amdgpu-va-drivers, rocprofiler-register
 Description: High perf video decode SDK for AMD GPUs. Rocdecode library and license.txt
 Homepage: https://github.com/ROCm/rocDecode
 Maintainer: rocdecode Support <mivisionx.support@amd.com>
 Package: rocdecode
 Priority: optional
 Section: devel
 Version: 1.4.0.70002-local
 Installed-Size: 1562

```
* rocdecode development

```
dpkg -I rocdecode-dev
rocdecode-dev_1.4.0.70002-local_amd64.deb     rocdecode-devel-1.4.0.70002-local.x86_64.rpm  
root@santiago:~/build# dpkg -I rocdecode-dev
rocdecode-dev_1.4.0.70002-local_amd64.deb     rocdecode-devel-1.4.0.70002-local.x86_64.rpm  
root@santiago:~/build# dpkg -I rocdecode-dev_1.4.0.70002-local_amd64.deb 
 new debian package, version 2.0.
 size 92027060 bytes: control archive=2281 bytes.
     481 bytes,    11 lines      control              
    5765 bytes,    58 lines      md5sums              
 Architecture: amd64
 Depends:  rocdecode, rocm-hip-runtime-dev, libva-dev (>= 2.16.0) | libva-amdgpu-dev, pkg-config, libavcodec-dev, libavformat-dev, libavutil-dev, libstdc++-12-dev
 Description: High perf video decode SDK for AMD GPUs. Rocdecode header files, and samples
 Homepage: https://github.com/ROCm/rocDecode
 Maintainer: rocdecode Support <mivisionx.support@amd.com>
 Package: rocdecode-dev
 Priority: optional
 Section: devel
 Version: 1.4.0.70002-local
 Installed-Size: 91150

```

kiritigowda commented:
@Mystro256 -- need you attention in SLES/RHEL 9

```
cat /etc/os-release 
NAME="SLES"
VERSION="15-SP6"
VERSION_ID="15.6"
PRETTY_NAME="SUSE Linux Enterprise Server 15 SP6"
ID="sles"
ID_LIKE="suse"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:15:sp6"
DOCUMENTATION_URL="https://documentation.suse.com/"

 vainfo --display drm
Trying display: drm
libva info: VA-API version 1.16.0
libva info: Trying to open /opt/amdgpu/lib64/dri/radeonsi_drv_video.so
libva error: /opt/amdgpu/lib64/dri/radeonsi_drv_video.so has no function __vaDriverInit_1_0
libva info: va_openDriver() returns -1
vaInitialize failed with error code -1 (unknown libva error),exit
```
